packages/lib/language/service.ts

```diff
@@ -110,25 +113,28 @@ export const getSurveysUsingGivenLanguage = reactCache(async (languageId: string
   }
 });
 
-export const deleteLanguage = async (environmentId: string, languageId: string): Promise<TLanguage> => {
+export const deleteLanguage = async (languageId: string, productId: string): Promise<TLanguage> => {
   try {
-    validateInputs([languageId, ZId]);
-
+    validateInputs([languageId, ZId], [productId, ZId]);
+    const product = await getProduct(productId);
+    if (!product) throw new ResourceNotFoundError("Product not found", productId);
     const prismaLanguage = await prisma.language.delete({
       where: { id: languageId },
       select: { ...languageSelect, surveyLanguages: { select: { surveyId: true } } },
     });
 
-    productCache.revalidate({
-      id: prismaLanguage.productId,
-      environmentId,
-    });
+    product.environments.forEach((environment) => {
+      productCache.revalidate({
+        id: prismaLanguage.productId,
+        environmentId: environment.id,
+      });
 
-    // revalidate cache of all connected surveys
-    prismaLanguage.surveyLanguages.forEach((surveyLanguage) => {
-      surveyCache.revalidate({
-        id: surveyLanguage.surveyId,
-        environmentId,
+      // revalidate cache of all connected surveys
+      prismaLanguage.surveyLanguages.forEach((surveyLanguage) => {
+        surveyCache.revalidate({
+          id: surveyLanguage.surveyId,
+          environmentId: environment.id,
+        });
```

Comment:
Can we please optimize this? 
This revalidate handler will run twice for the same set of surveys.